### DISCUSSION TITLE
OSX M2 Support Still Non-existent?

### DISCUSSION THREAD

**Discussion by JJHelly** [pyswmm/pyswmm#556] [General]
Aloha.

I have managed to get swmm524 to run on OSX but can't do anything with the *.out files.  I have tried running pyswmm under python3.12 in a virtual environment but seem to get segfaults.  

I'm not sure what the cause is but I suspect it has to do with pyswmm not actually supported for OSX.  For example, there's a place for a link on the tutorial for installation there is nothing there; similarly for Debian.

Am I correct in concluding that OSX is not a supported platform nor Debian?

**Comment by karosc** [comment]
Both osx-arm64 and Linux-aarch64 are supported as well as the amd64 architectures are supported up through python 3.12. See the pypi link below for the binary listings.

https://pypi.org/project/swmm-toolkit/#files

If you're getting segfaults reading your output file, there might be a bug in your code or in pyswmm. If you can post code and model files that reproduce the error, we can support debugging.

**Comment by JJHelly** [comment]
Great to hear.  Mahalo for the reply.  I will follow the link.

J.
----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Oct 31, 2024, at 15:24, Constantine Karos ***@***.***> wrote:



Both osx-arm64 and Linux-aarch64 are supported as well as the amd64 architectures are supported up through python 3.12. See the pypi link below for the binary listings.

https://pypi.org/project/swmm-toolkit/#files<https://urldefense.com/v3/__https://pypi.org/project/swmm-toolkit/*files__;Iw!!Mih3wA!HPRkaVG4unvSZzYdIJ_QbxYuC0GJB4XdddwunzxGoUvmBL0_Ix5jMDe1DEhPH32EZ11FLqpCabXYr2uhglazssrQ$>

If you're getting segfaults reading your output file, there might be a bug in your code or in pyswmm. If you can post code and model files that reproduce the error, we can support debugging.

**Comment by JJHelly** [comment]
Aloha.

No joy, I'm afraid.  I'm installed in a venv and installed pyswmm and swmm-toolkit apparently successfully.  I'm assuming the swmm5 binary is bundled with pyswmm.  Is that true?


zsh: segmentation fault  ../bin/python3.12 Latte_JJH.py


=========================================================

Below is Latte_JJH.py.  I'm trying to adapt Latte.py obiously.  I'm assuming the sim step is what's failing?


I'm on OSX Sequoia 15.0.1 and a mac studio M2

Mahalo for your help.


Perhaps there's a test I can try?

J.

=========================================================


'''

PySWMM Latte Art Code

Author: Bryant McDonnell

Version: 1

Date: Nov 17, 2022

'''

import matplotlib.pyplot as plt

import matplotlib.dates as mdates

from pyswmm import Simulation, Nodes, Links, Output

from swmm.toolkit.shared_enum import SubcatchAttribute, NodeAttribute, LinkAttribute


with Simulation(r'./3086_Post_Alt1.inp') as sim:

    Node21 = Nodes(sim)["POC-1"]


    # Initialize Lists for storing data

    time_stamps = []

    node_head = []

    link_flow = []


    sim.step_advance(300)

    # Launch a simulation!

    for ind, step in enumerate(sim):

#        print(sim.current_time)

        time_stamps.append(sim.current_time)

        node_head.append(Node21.head)


with Output('./3086_Post_Alt1.inp.JJH.out') as out:

    node_head_outfile = out.node_series('POC-1', NodeAttribute.HYDRAULIC_HEAD)


# https://matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.figure.html#matplotlib.pyplot.figure

fig = plt.figure(figsize=(8,4), dpi=200) #Inches Width, Height

fig.suptitle("Node 21 Head and Link 15 Flow from simulation and output")

# Plot from the results compiled during simulation time

axis_1 = fig.add_subplot(2,1,1)

axis_1.plot(time_stamps, node_head, '-g', label="Running Sim")

# Plot from the output file

x = node_head_outfile.keys()

y = [node_head_outfile[key] for key in node_head_outfile.keys()]

axis_1.plot(x, y, ':b', label="Output File")

axis_1.set_ylabel("Head (ft)")

#axis_1.get_xticklabels().set_visible(False) # turns off the labels

axis_1.grid("xy")

axis_1.legend()

# Second Axis

axis_2 = fig.add_subplot(2,1,2, sharex=axis_1)

axis_2.plot(time_stamps, link_flow, ls='-', color = 'g')

x = link_flow_outfile.keys()

y = [link_flow_outfile[key] for key in link_flow_outfile.keys()]

axis_2.plot(x, y, ':b', label="Output File")

axis_2.set_ylabel("Flow (CFS)")

axis_2.xaxis.set_major_formatter(mdates.DateFormatter('%m-%d %Hh'))

axis_2.grid("xy")


fig.autofmt_xdate()

plt.tight_layout()

plt.savefig("TEST.PNG")

plt.show()




----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Oct 31, 2024, at 15:24, Constantine Karos ***@***.***> wrote:



Both osx-arm64 and Linux-aarch64 are supported as well as the amd64 architectures are supported up through python 3.12. See the pypi link below for the binary listings.

https://pypi.org/project/swmm-toolkit/#files<https://urldefense.com/v3/__https://pypi.org/project/swmm-toolkit/*files__;Iw!!Mih3wA!HPRkaVG4unvSZzYdIJ_QbxYuC0GJB4XdddwunzxGoUvmBL0_Ix5jMDe1DEhPH32EZ11FLqpCabXYr2uhglazssrQ$>

If you're getting segfaults reading your output file, there might be a bug in your code or in pyswmm. If you can post code and model files that reproduce the error, we can support debugging.

**Comment by karosc** [comment]
@JJHelly, I don't have an M2 mac, but I did try running your code. There were a few missing lines, but I think I was able to fill in the gaps. You also did not post your inp file, so I downloaded the one from the Latte example on the pyswmm website and used that one. See if this runs on your mac. The python code is shown below, but also in the attached zipfile along with the Example1.inp file.

```python
import matplotlib.pyplot as plt

import matplotlib.dates as mdates


from pyswmm import Simulation, Nodes, Links, Output

from pyswmm.links import Link
from swmm.toolkit.shared_enum import SubcatchAttribute, NodeAttribute, LinkAttribute


with Simulation(r"./Example1.inp") as sim:

    Node21 = Nodes(sim)["21"]
    Link15 = Links(sim)["15"]

    # Initialize Lists for storing data
    time_stamps = []
    node_head = []
    link_flow = []
    sim.step_advance(300)

    # Launch a simulation!
    for ind, step in enumerate(sim):
        time_stamps.append(sim.current_time)
        node_head.append(Node21.head)
        link_flow.append(Link15.flow)


with Output("./Example1.out") as out:
    node_head_outfile = out.node_series("21", NodeAttribute.HYDRAULIC_HEAD)
    link_flow_outfile = out.link_series("15", LinkAttribute.FLOW_RATE)


# matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.figure.html#matplotlib.pyplot.figure

fig = plt.figure(figsize=(8, 4), dpi=200)  # Inches Width, Height
fig.suptitle("Node 21 Head and Link 15 Flow from simulation and output")

# Plot from the results compiled during simulation time
axis_1 = fig.add_subplot(2, 1, 1)
axis_1.plot(time_stamps, node_head, "-g", label="Running Sim")

# Plot from the output file
x = node_head_outfile.keys()
y = [node_head_outfile[key] for key in node_head_outfile.keys()]
axis_1.plot(x, y, ":b", label="Output File")
axis_1.set_ylabel("Head (ft)")
# axis_1.get_xticklabels().set_visible(False) # turns off the labels
axis_1.grid("xy")
axis_1.legend()

# Second Axis
axis_2 = fig.add_subplot(2, 1, 2, sharex=axis_1)
axis_2.plot(time_stamps, link_flow, ls="-", color="g")

x = link_flow_outfile.keys()
y = [link_flow_outfile[key] for key in link_flow_outfile.keys()]

axis_2.plot(x, y, ":b", label="Output File")
axis_2.set_ylabel("Flow (CFS)")
axis_2.xaxis.set_major_formatter(mdates.DateFormatter("%m-%d %Hh"))
axis_2.grid("xy")


fig.autofmt_xdate()

plt.tight_layout()
plt.savefig("TEST.PNG")
plt.show()
```

![TEST](https://github.com/user-attachments/assets/48ee0f8d-fa22-4352-942a-58cc75309cc8)
[latte.zip](https://github.com/user-attachments/files/17599775/latte.zip)

**Comment by JJHelly** [comment]
Mahalo.  I will give it a try.

I posted the *.inp and *.dat files in a subsequent apologia message.

J.
------------------------------------------------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Nov 1, 2024, at 02:57, Constantine Karos ***@***.***> wrote:



@JJHelly<https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8Rei49TX$>, I don't have an M2 mac, but I did try running your code. There were a few missing lines, but I think I was able to fill in the gaps. You also did not post your inp file, so I downloaded the one from the Latte example on the pyswmm website and used that one. See if this runs on your mac. The python code is shown below, but also in the attached zipfile along with the Example1.inp file.

import matplotlib.pyplot as plt

import matplotlib.dates as mdates
[latte.zip](https://github.com/user-attachments/files/17599764/latte.zip)

from pyswmm import Simulation, Nodes, Links, Output

from pyswmm.links import Link
from swmm.toolkit.shared_enum import SubcatchAttribute, NodeAttribute, LinkAttribute


with Simulation(r"./Example1.inp") as sim:

    Node21 = Nodes(sim)["21"]
    Link15 = Links(sim)["15"]

    # Initialize Lists for storing data
    time_stamps = []
    node_head = []
    link_flow = []
    sim.step_advance(300)

    # Launch a simulation!
    for ind, step in enumerate(sim):
        time_stamps.append(sim.current_time)
        node_head.append(Node21.head)
        link_flow.append(Link15.flow)


with Output("./Example1.out") as out:
    node_head_outfile = out.node_series("21", NodeAttribute.HYDRAULIC_HEAD)
    link_flow_outfile = out.link_series("15", LinkAttribute.FLOW_RATE)


# matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.figure.html#matplotlib.pyplot.figure

fig = plt.figure(figsize=(8, 4), dpi=200)  # Inches Width, Height
fig.suptitle("Node 21 Head and Link 15 Flow from simulation and output")

# Plot from the results compiled during simulation time
axis_1 = fig.add_subplot(2, 1, 1)
axis_1.plot(time_stamps, node_head, "-g", label="Running Sim")

# Plot from the output file
x = node_head_outfile.keys()
y = [node_head_outfile[key] for key in node_head_outfile.keys()]
axis_1.plot(x, y, ":b", label="Output File")
axis_1.set_ylabel("Head (ft)")
# axis_1.get_xticklabels().set_visible(False) # turns off the labels
axis_1.grid("xy")
axis_1.legend()

# Second Axis
axis_2 = fig.add_subplot(2, 1, 2, sharex=axis_1)
axis_2.plot(time_stamps, link_flow, ls="-", color="g")

x = link_flow_outfile.keys()
y = [link_flow_outfile[key] for key in link_flow_outfile.keys()]

axis_2.plot(x, y, ":b", label="Output File")
axis_2.set_ylabel("Flow (CFS)")
axis_2.xaxis.set_major_formatter(mdates.DateFormatter("%m-%d %Hh"))
axis_2.grid("xy")


fig.autofmt_xdate()

plt.tight_layout()
plt.savefig("TEST.PNG")
plt.show()

TEST.PNG (view on web)<https://urldefense.com/v3/__https://github.com/user-attachments/assets/48ee0f8d-fa22-4352-942a-58cc75309cc8__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8Y0Py_LH$>
latte.zip<https://urldefense.com/v3/__https://github.com/user-attachments/files/17599775/latte.zip__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8WRl_fU1$>

**Comment by JJHelly** [comment]
Aloha K.

Well, I'm stymied on every front.  I'm trying to start from scratch in a venv and, even though pip says it's loaded, pyswmm can't be imported.  Here is the series of commands and their outcome.

Please advise what is wrong/missing.  This seems to follow every tutorial I can find but little progress with python overall.

Very much appreciate your help.

================================= command history follows =========================


1023  python -m venv venv

 1024  source venv/bin/activate

 1025  pip

 1026  pip install matplotlib

 1027  pip install swmm-toolkit

 1028  pip install --upgrade pip

 1029  pip install pyswmm

 1030  ls

 1031  cd Espresso

 1032  ls

 1033  python Espresso.py

(venv) ***@***.*** Espresso % python Espresso.py

(venv) ***@***.*** Espresso % cd ..

(venv) ***@***.*** ~ % python ./Espresso/Espresso.py

Traceback (most recent call last):

  File "/Users/hellyj/./Espresso/Espresso.py", line 7, in <module>

    from pyswmm import Simulation

ModuleNotFoundError: No module named 'pyswmm'

(venv) ***@***.*** ~ % python

Python 3.12.7 (main, Oct  5 2024, 01:39:55) [Clang 16.0.0 (clang-1600.0.26.3)] on darwin

Type "help", "copyright", "credits" or "license" for more information.

>>> import pyswmm

Traceback (most recent call last):

  File "<stdin>", line 1, in <module>

ModuleNotFoundError: No module named 'pyswmm'

>>> ^D

(venv) ***@***.*** ~ % pip --version

pip 24.3.1 from /Users/hellyj/venv/lib/python3.12/site-packages/pip (python 3.12)

(venv) ***@***.*** ~ % pip install pyswmm

Requirement already satisfied: pyswmm in ./venv/lib/python3.12/site-packages (2.0.1)

Requirement already satisfied: swmm-toolkit>=0.9.0 in ./venv/lib/python3.12/site-packages (from pyswmm) (0.15.5)

Requirement already satisfied: julian>=0.14 in ./venv/lib/python3.12/site-packages (from pyswmm) (0.14)

Requirement already satisfied: aenum>=3.1.11 in ./venv/lib/python3.12/site-packages (from pyswmm) (3.1.11)

Requirement already satisfied: packaging in ./venv/lib/python3.12/site-packages (from pyswmm) (24.1)

(venv) ***@***.*** ~ % python ./Espresso/Espresso.py

Traceback (most recent call last):

  File "/Users/hellyj/./Espresso/Espresso.py", line 7, in <module>

    from pyswmm import Simulation

ModuleNotFoundError: No module named 'pyswmm'

(venv) ***@***.*** ~ %

----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Nov 1, 2024, at 02:57, Constantine Karos ***@***.***> wrote:



@JJHelly<https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8Rei49TX$>, I don't have an M2 mac, but I did try running your code. There were a few missing lines, but I think I was able to fill in the gaps. You also did not post your inp file, so I downloaded the one from the Latte example on the pyswmm website and used that one. See if this runs on your mac. The python code is shown below, but also in the attached zipfile along with the Example1.inp file.

import matplotlib.pyplot as plt

import matplotlib.dates as mdates
[latte.zip](https://github.com/user-attachments/files/17599764/latte.zip)

from pyswmm import Simulation, Nodes, Links, Output

from pyswmm.links import Link
from swmm.toolkit.shared_enum import SubcatchAttribute, NodeAttribute, LinkAttribute


with Simulation(r"./Example1.inp") as sim:

    Node21 = Nodes(sim)["21"]
    Link15 = Links(sim)["15"]

    # Initialize Lists for storing data
    time_stamps = []
    node_head = []
    link_flow = []
    sim.step_advance(300)

    # Launch a simulation!
    for ind, step in enumerate(sim):
        time_stamps.append(sim.current_time)
        node_head.append(Node21.head)
        link_flow.append(Link15.flow)


with Output("./Example1.out") as out:
    node_head_outfile = out.node_series("21", NodeAttribute.HYDRAULIC_HEAD)
    link_flow_outfile = out.link_series("15", LinkAttribute.FLOW_RATE)


# matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.figure.html#matplotlib.pyplot.figure

fig = plt.figure(figsize=(8, 4), dpi=200)  # Inches Width, Height
fig.suptitle("Node 21 Head and Link 15 Flow from simulation and output")

# Plot from the results compiled during simulation time
axis_1 = fig.add_subplot(2, 1, 1)
axis_1.plot(time_stamps, node_head, "-g", label="Running Sim")

# Plot from the output file
x = node_head_outfile.keys()
y = [node_head_outfile[key] for key in node_head_outfile.keys()]
axis_1.plot(x, y, ":b", label="Output File")
axis_1.set_ylabel("Head (ft)")
# axis_1.get_xticklabels().set_visible(False) # turns off the labels
axis_1.grid("xy")
axis_1.legend()

# Second Axis
axis_2 = fig.add_subplot(2, 1, 2, sharex=axis_1)
axis_2.plot(time_stamps, link_flow, ls="-", color="g")

x = link_flow_outfile.keys()
y = [link_flow_outfile[key] for key in link_flow_outfile.keys()]

axis_2.plot(x, y, ":b", label="Output File")
axis_2.set_ylabel("Flow (CFS)")
axis_2.xaxis.set_major_formatter(mdates.DateFormatter("%m-%d %Hh"))
axis_2.grid("xy")


fig.autofmt_xdate()

plt.tight_layout()
plt.savefig("TEST.PNG")
plt.show()

TEST.PNG (view on web)<https://urldefense.com/v3/__https://github.com/user-attachments/assets/48ee0f8d-fa22-4352-942a-58cc75309cc8__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8Y0Py_LH$>
latte.zip<https://urldefense.com/v3/__https://github.com/user-attachments/files/17599775/latte.zip__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8WRl_fU1$>

**Comment by karosc** [comment]
Oh man, it looks like you are having trouble setting up you python environment properly. It's really hard for us to help with general tech support issues like that from a github thread like this. I really recommend you look into having a conversation with either https://chatgpt.com or https://claude.ai. Copying an pasting your commands and errors into those tools will likely get you through this problem much faster than I can and with a better experience for you.

**Comment by JJHelly** [comment]
Interesting suggestion. Mahalo.  These kinds of problems are why I tend to avoid python but I keep hoping.


J.
------------------------------------------------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Nov 1, 2024, at 16:27, Constantine Karos ***@***.***> wrote:



Oh man, it looks like you are having trouble setting up you python environment properly. It's really hard for us to help with general tech support issues like that from a github thread like this. I really recommend you look into having a conversation with either https://chatgpt.com<https://urldefense.com/v3/__https://chatgpt.com__;!!Mih3wA!GoKW5maLwmFuNWG9mWqD2CXwXHZFD6AZoYkSwZF1IKix6fzQPjikkeBhPT34ddKhsEwp5Du85HlXWg2P_nvJvGuU$> or https://claude.ai<https://urldefense.com/v3/__https://claude.ai__;!!Mih3wA!GoKW5maLwmFuNWG9mWqD2CXwXHZFD6AZoYkSwZF1IKix6fzQPjikkeBhPT34ddKhsEwp5Du85HlXWg2P_nWUJCbK$>. Copying an pasting your commands and errors into those tools will likely get you through this problem much faster than I can and with a better experience for you.

**Comment by JJHelly** [comment]
Ok. I danced in circles and bowed east and west and managed to get python312 to work on OSX using macports.  I also was able to run your example (latte.zip) and reproduce my segfault.  So it looks like there is something wrong with my code.
----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Nov 1, 2024, at 02:57, Constantine Karos ***@***.***> wrote:



@JJHelly<https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8Rei49TX$>, I don't have an M2 mac, but I did try running your code. There were a few missing lines, but I think I was able to fill in the gaps. You also did not post your inp file, so I downloaded the one from the Latte example on the pyswmm website and used that one. See if this runs on your mac. The python code is shown below, but also in the attached zipfile along with the Example1.inp file.

import matplotlib.pyplot as plt

import matplotlib.dates as mdates
[latte.zip](https://github.com/user-attachments/files/17599764/latte.zip)

from pyswmm import Simulation, Nodes, Links, Output

from pyswmm.links import Link
from swmm.toolkit.shared_enum import SubcatchAttribute, NodeAttribute, LinkAttribute


with Simulation(r"./Example1.inp") as sim:

    Node21 = Nodes(sim)["21"]
    Link15 = Links(sim)["15"]

    # Initialize Lists for storing data
    time_stamps = []
    node_head = []
    link_flow = []
    sim.step_advance(300)

    # Launch a simulation!
    for ind, step in enumerate(sim):
        time_stamps.append(sim.current_time)
        node_head.append(Node21.head)
        link_flow.append(Link15.flow)


with Output("./Example1.out") as out:
    node_head_outfile = out.node_series("21", NodeAttribute.HYDRAULIC_HEAD)
    link_flow_outfile = out.link_series("15", LinkAttribute.FLOW_RATE)


# matplotlib.org/3.1.0/api/_as_gen/matplotlib.pyplot.figure.html#matplotlib.pyplot.figure

fig = plt.figure(figsize=(8, 4), dpi=200)  # Inches Width, Height
fig.suptitle("Node 21 Head and Link 15 Flow from simulation and output")

# Plot from the results compiled during simulation time
axis_1 = fig.add_subplot(2, 1, 1)
axis_1.plot(time_stamps, node_head, "-g", label="Running Sim")

# Plot from the output file
x = node_head_outfile.keys()
y = [node_head_outfile[key] for key in node_head_outfile.keys()]
axis_1.plot(x, y, ":b", label="Output File")
axis_1.set_ylabel("Head (ft)")
# axis_1.get_xticklabels().set_visible(False) # turns off the labels
axis_1.grid("xy")
axis_1.legend()

# Second Axis
axis_2 = fig.add_subplot(2, 1, 2, sharex=axis_1)
axis_2.plot(time_stamps, link_flow, ls="-", color="g")

x = link_flow_outfile.keys()
y = [link_flow_outfile[key] for key in link_flow_outfile.keys()]

axis_2.plot(x, y, ":b", label="Output File")
axis_2.set_ylabel("Flow (CFS)")
axis_2.xaxis.set_major_formatter(mdates.DateFormatter("%m-%d %Hh"))
axis_2.grid("xy")


fig.autofmt_xdate()

plt.tight_layout()
plt.savefig("TEST.PNG")
plt.show()

TEST.PNG (view on web)<https://urldefense.com/v3/__https://github.com/user-attachments/assets/48ee0f8d-fa22-4352-942a-58cc75309cc8__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8Y0Py_LH$>
latte.zip<https://urldefense.com/v3/__https://github.com/user-attachments/files/17599775/latte.zip__;!!Mih3wA!E6epcqmww3lpYMOPsT54pvD5e5U3-ueWAmxILRByjqMJEG8suFFbwbBDa9pijj7zFIzaD6gVhYy6YuY_8WRl_fU1$>

**Comment by JJHelly** [comment]
I figured out that the seg fault was due to the fact that the Output file did not exist.  I made a mistake in the name.  Seems like pyswmm should be trapping this kind of error and handling it better or is this something deep in python?

Your code was quite helpful in providing a cross-reference.  Mahalo.

J.
----------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Nov 1, 2024, at 16:27, Constantine Karos ***@***.***> wrote:



Oh man, it looks like you are having trouble setting up you python environment properly. It's really hard for us to help with general tech support issues like that from a github thread like this. I really recommend you look into having a conversation with either https://chatgpt.com<https://urldefense.com/v3/__https://chatgpt.com__;!!Mih3wA!GoKW5maLwmFuNWG9mWqD2CXwXHZFD6AZoYkSwZF1IKix6fzQPjikkeBhPT34ddKhsEwp5Du85HlXWg2P_nvJvGuU$> or https://claude.ai<https://urldefense.com/v3/__https://claude.ai__;!!Mih3wA!GoKW5maLwmFuNWG9mWqD2CXwXHZFD6AZoYkSwZF1IKix6fzQPjikkeBhPT34ddKhsEwp5Du85HlXWg2P_nWUJCbK$>. Copying an pasting your commands and errors into those tools will likely get you through this problem much faster than I can and with a better experience for you.

**Reply by bemcdonnell** [reply]
@JJHelly , glad you got it all sorted out! As for error checking, those types of things can always be improved. If you have ideas or code that you’d be willing to share to help this project become better, we would be more than happy to work with you to integrate those.

**Comment by JJHelly** [comment]
Aloha.

Mahalo for all your hard work.  This is a really nice tool that fills an important gap.  I've tried to get swmmr to work but it seems to have problems with the C++/R API in newers osx versions.

To the case in point, I'm not a python programmer but the typical approach to these kinds of call to open files do an existence check first and throw an error if the target does not exist.  I think what's happening is that the existing code refers to an uninitialized variable that points off to some random place in memory.

I'll try to look at your code for the call and see if I can see what's going on but, as I say, I'm not a python programmer.

J.
------------------------------------------------------------------------------------------------------------------------------------------------------------
John Helly / San Diego Supercomputer Center / Scripps Institution of Oceanography
https://www.sdsc.edu/~hellyj / 808 205 9882 / 760 8408660

On Nov 2, 2024, at 03:41, Bryant E. McDonnell ***@***.***> wrote:



@JJHelly<https://urldefense.com/v3/__https://github.com/JJHelly__;!!Mih3wA!HyihYhobTmCIdwG-ubFqHmxXBMxM38QAWgnVfheqzof2boHLxRl3HyIP8Nem2tNB8mETfEY4ddl5nqqzWvTUlu3N$> , glad you got it all sorted out! As for error checking, those types of things can always be improved. If you have ideas or code that you’d be willing to share to help this project become better, we would be more than happy to work with you to integrate those.
